core/common/src/main/java/alluxio/collections/IndexedSet.java

```diff
@@ -176,26 +248,42 @@ public boolean add(T object) {
         return false;
       }
 
-      // Update the indexes.
-      for (Map.Entry<FieldIndex<T>, ConcurrentHashMap<Object, ConcurrentHashSet<T>>> fieldInfo :
-          mIndexMap.entrySet()) {
-        // For this field, retrieve the value to index
-        Object fieldValue = fieldInfo.getKey().getFieldValue(object);
-        // Get the index for this field
-        ConcurrentHashMap<Object, ConcurrentHashSet<T>> index = fieldInfo.getValue();
-        ConcurrentHashSet<T> objSet = index.get(fieldValue);
-        if (objSet == null) {
-          index.putIfAbsent(fieldValue, new ConcurrentHashSet<T>());
-          objSet = index.get(fieldValue);
+      if (mIndexMapUnique != null) {
```

Comment:
to be exact, not the for loop but the contents of the for loop, so this should simply become:

```
for (FieldIndex<T> index : mIndices) {
  index.put(object)
}
```